evap/static/ts/src/infobox.ts

```diff
@@ -0,0 +1,40 @@
+declare const bootstrap: typeof import("bootstrap");
+
+import { selectOrError } from "./utils.js";
+
+const OPEN_CLOSE_TIMEOUT = 2000;
+
+export class InfoboxLogic {
+    private readonly infobox: HTMLDivElement;
+    private readonly closeButton: HTMLButtonElement;
+    private readonly storageKey: string;
+
+    constructor(infobox_id: string) {
+        this.infobox = selectOrError("#" + infobox_id);
+        this.closeButton = selectOrError("#" + infobox_id + " .callout-infobox-close");
+        this.storageKey = "infobox_" + infobox_id;
+    }
+
+    public attach = (): void => {
+        // close the infobox and save state
+        this.closeButton.addEventListener("click", async event => {
+            this.infobox.classList.add("closing");
+            setTimeout(() => {
+                this.infobox.classList.replace("closing", "closed");
+            }, OPEN_CLOSE_TIMEOUT);
```

Comment:
I think we've talked about this, but I've forgotten, so maybe we should document it somewhere: Why do we need seperate `closing` and `closed` classes? Doesn't CSS work that way that it simply animates towards the target state and then stays there? Then, `open` and `closed` should suffice.

Reply:
Now, the infobox can be reopened immediately, even when not completely closed yet. So, the background changes, which might be annoying. What does @janno42 think?

Reply:
(we discussed in person, and decided that we do not want the "you can click on this"-hover-behavior while the infobox is still closing, so we need an additional class during transition. For javascript reasons, we also can't get the timeout from the element in a nice way here)